kombu/transport/redis.py

```diff
@@ -384,7 +384,7 @@ class Channel(virtual.Channel):
     unacked_restore_limit = None
     visibility_timeout = 3600   # 1 hour
     priority_steps = PRIORITY_STEPS
-    socket_timeout = None
+    socket_timeout = 60
```

Comment:
Setting this to 60 so that if its hanging here, it shouldnt hang for more than 60 seconds. If your application requirements are different, this might have to be a smaller number.